src/Elastic.Apm/Helpers/StacktraceHelper.cs

```diff
@@ -53,21 +56,42 @@ internal static class StacktraceHelper
 			{
 				foreach (var frame in frames)
 				{
-					var fileName = frame?.GetMethod()
+					var className = frame?.GetMethod()
 						?.DeclaringType?.FullName; //see: https://github.com/elastic/apm-agent-dotnet/pull/240#discussion_r289619196
 
 					var functionName = GetRealMethodName(frame?.GetMethod());
+					var fileName = frame?.GetFileName();
 
 					logger.Trace()?.Log("{MethodName}, {lineNo}", functionName, frame?.GetFileLineNumber());
 
-					retVal.Add(new CapturedStackFrame
+					var capturedStackFrame = new CapturedStackFrame
 					{
 						Function = functionName ?? "N/A",
-						FileName = string.IsNullOrWhiteSpace(fileName) ? "N/A" : fileName,
 						Module = frame?.GetMethod()?.ReflectedType?.Assembly.FullName,
 						LineNo = frame?.GetFileLineNumber() ?? 0,
 						AbsPath = frame?.GetFileName() // optional property
-					});
+					};
+
+					if (serverInfo.ServerVersionQueried && serverInfo.Version != null && serverInfo.Version < V710)
```

Comment:
Thoughts on having an `IStackTrace[Generator|Transformer]` with two different implementations to perform the relevant changes to `CapturedStackFrame`?

Reply:
For the changes in this PR I would not do it - long term this `if` block will go away. We can completely remove this part once 7.9 is out of support.

But in general I agree there is a potential to better structure the stack trace capturing logic. Happy to make follow ups, but for solving the current problem from this PR I think this is fine.